modules/affile/wildcard-source.c

```diff
@@ -77,11 +77,17 @@ _remove_file_reader(FileReader *reader, gpointer user_data)
     }
   log_pipe_unref(&reader->super);
 
-  gchar *full_path = pending_file_list_pop(self->waiting_list);
-  if (full_path)
+  gchar *full_path = NULL;
+  while ((full_path = pending_file_list_pop(self->waiting_list)))
     {
+      if (g_hash_table_lookup_extended(self->file_readers, full_path, NULL, NULL))
+        {
+          g_free(full_path);
+          continue;
```

Comment:
Is it OK to remove (pop) the path from the waiting list when we skip creating a reader from it?
I mean the old and new file with the same name can be different.
We might read an old deleted/renamed file with its FD kept open, but we do want to schedule the new one after we reach EOF in the old one.

Reply:
you are right, i've added a commit to it, please check.

Reply:
Thanks!